### PR TITLE
Feat/crud role mapping

### LINKED ISSUE #1153: CRUD operations for Role mapping
|Wazuh| Splunk |
| --- | --- |
| 4.2.x| 8.1.4 / 8.2.2 |

**Description**
Add CRUD operations to manage role mapping in the new Splunk Security section.

Tasks:

- [x] List Role Mappings.
- [x] Add Role Mapping.
- [x] Edit Role Mapping.
- [x] Delete Role Mapping.

### PR DESCRIPTION
**Pull request by eze9252**
Hi team,
This PR added new CRUD of Roles Mapping on the security section.
Closes #1153 

List Role Mapping: 

Captura de pantalla de 2021-11-03 18-41-04

Create Role Mapping:
Captura de pantalla de 2021-11-03 18-54-02

See and update Role Mapping:
Captura de pantalla de 2021-11-03 18-41-14

Delete Role Mapping:
Captura de pantalla de 2021-11-03 18-54-11

### CODE REVIEW

File: SplunkAppForWazuh/appserver/static/js/directives/wz-table/wz-table.html [hunk 1]
```diff
@@ -343,7 +343,53 @@
...
+              </div>              
+              <div layout="row" ng-if="path === '/security/rules'">
+                <md-button class="cancelBtn" type="button" ng-click="cancelRemoveRoleMapping()"><i aria-hidden='true'
+                    class='fa fa-fw fa-close'></i>
+                  Cancel</md-button>
+                <md-button class="agreeBtn" type="button" ng-click="confirmRemoveRoleMapping(item)"><i aria-hidden='true'
+                    class='fa fa-fw fa-check'></i> Confirm</md-button>
+              </div>
+            </div>
+            <td ng-if="isLookingGroup() " ng-click="$event.stopPropagation()"
```
Comment (gabiwassan) [inline SplunkAppForWazuh/appserver/static/js/directives/wz-table/wz-table.html:370]: I think this method `isLookingGroup()` is for Agents Groups

File: SplunkAppForWazuh/appserver/static/js/directives/wz-table/wz-table.html [hunk 2]
```diff
@@ -343,7 +343,53 @@
...
+              class="cursor-default action-btn-td">
+              <wz-svg ng-if="removingRoleMapping !== item.name"
+                ng-click="showConfirmRemoveRoleMapping($event, item); $event.stopPropagation()" icon="trash"
+                color="#396e3e" svg-tooltip="Remove entry">
+              </wz-svg>
+
+              <div ng-if="path === '/security/rules'" class="confirmEmbedBubble">
+                <div layout="row">
+                  <span class="font-size-12 wz-padding-left-8">{{item.name}} will be removed from
+                    this group</span>
```
Comment (gabiwassan) [inline SplunkAppForWazuh/appserver/static/js/directives/wz-table/wz-table.html:380]: This message is referred to groups not to Role Mapping 🤓

**Review by gabiwassan — CHANGES_REQUESTED** [review]
Just some simple comments, and please check to apply `prettier` for the style code.

**Comment by Desvelao** [comment]
**review**:

Testing:
- Table
  - List role mapping ✅ 
  **suggestion**: use `Roles` instead of `roles` for the column. Is there some limitation to do this?
  **suggestion**: use the role name instead of its ID in the role mapping table
  **suggestion**: the last column should be named `Actions`. See another security tabs.
  image
  - Manage columns visibility ✅ 
  - `action:edit` (open the editing view) ✅ 
  - `action:delete` ❌ 
    - Cancel ✅ 
    - Confirm ❌ 
    **issue**: The role mapping rule is deleted but there is an error when trying to list again and update the table 
- Add new role mapping
**question**: Should we add the rule JSON editor?
  - Role multi-select ✅
  - User multi-select ✅
  - `action:save` :x:
  **issue**: If I try to edit the role mapping an error is thrown and the save button keeps processing indefinitely: 
  image
Request:
 ``` 
<SPLUNK_HOST>:<SPLUNK_PORT><SPLUNK_BASE>/api/request
```
image

Response:
```json
{"error": "'error'"}
```

**Pull request merged**